File: Dataset/Leetcode/train/6/92.java
 class Solution {
    public String XXX(String s, int numRows) {

        if (numRows <= 1) return s;

        int left = (numRows - 1) << 1;
        int right = 0;
        int row = 0;
        char[] newS = new char[s.length()];

        for (int i = 0, j = 0; i < s.length();) {

            if (left != 0 && j < s.length()) {
                newS[i++] = s.charAt(j);
                j += left;
            }

            if (right != 0 && j < s.length()) {
                newS[i++] = s.charAt(j);
                j += right;
            }

            if (j >= s.length()) {
                left -= 2;
                right += 2;
                row++;
                j = row;
            }
        }

        return new String(newS);
    }
}

File: Dataset/Leetcode/test/70/92.js
    var XXX = function(n) {
        if (n === 1 || n === 2) {
            return n
        }

        if (XXX[n]) {
            return XXX[n]
        }

        XXX[n] = XXX(n - 1) + XXX(n - 2)
        return  XXX[n]
    };

